solvers/lightning.py

```diff
@@ -0,0 +1,60 @@
+import warnings
+from benchopt import BaseSolver
+from benchopt import safe_import_context
+
+
+with safe_import_context() as import_ctx:
+    import numpy as np
+    from lightning.regression import SDCARegressor
+    from sklearn.exceptions import ConvergenceWarning
+
+
+class Solver(BaseSolver):
+    name = 'lightning'
+
+    install_cmd = 'conda'
+    requirements = ['sklearn-contrib-lightning']
+    references = [
+        'M. Blondel, K. Seki and K. Uehara, '
+        '"Block coordinate descent algorithms for large-scale sparse '
+        'multiclass classification" '
+        'Mach. Learn., vol. 93, no. 1, pp. 31-52 (2013)'
+    ]
+
+    def set_objective(self, X, y, l1_ratio, lmbda, fit_intercept):
+        self.X, self.y, self.l1_ratio, self.lmbda = X, y, l1_ratio, lmbda
+        self.fit_intercept = fit_intercept
+
+        self.clf = SDCARegressor(alpha=self.lmbda,
+                                 l1_ratio=self.l1_ratio, tol=0)
+        warnings.filterwarnings('ignore', category=ConvergenceWarning)
+
+        self.run(1)
+
+    @staticmethod
+    def get_next(stop_val):
+        return stop_val + 1
+
+    def run(self, n_iter):
+        if n_iter == 0:
+            self.coef = np.zeros(self.X.shape[1] + self.fit_intercept)
+        else:
+            self.clf.max_iter = n_iter
+            X, y = self.X, self.y
+
+            if self.fit_intercept:
```

Comment:
I used the trick of equivalence between lasso + intercept and lasso center data to add support of intercept into the ElasticNet.
(It breaks the sparsity data)

Reply:
The elastic net is not equivariant to such changes because of the L2 penalty. If you manage to make it work with the Lasso equivalence I'm interested but I don't think it's possible.
IMO you should skip this solver when fit_intercept=False.